packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -121,11 +122,22 @@ export class BundlingDockerImage {
       throw new Error('Failed to extract image ID from Docker build output');
     }
 
-    return new BundlingDockerImage(match[1]);
+    const hash = FileSystem.fingerprint(path, { extraHash: JSON.stringify(options) });
```

Comment:
Add some commentary on what you are doing here

Reply:
I've added some commentary here.